eth/protocols/eth/peer.go

```diff
@@ -141,22 +143,23 @@ func (p *Peer) Version() uint {
 }
 
 // Head retrieves the current head hash and head number of the peer.
-func (p *Peer) Head() (hash common.Hash, number uint64, receivedAt time.Time) {
+func (p *Peer) Head() (hash common.Hash, number *big.Int, entropy *big.Int, receivedAt time.Time) {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 
 	copy(hash[:], p.head[:])
-	return hash, p.number, p.receivedHeadAt
+	return hash, p.number, p.entropy, p.receivedHeadAt
 }
 
 // SetHead updates the head hash and head number of the peer.
-func (p *Peer) SetHead(hash common.Hash, number uint64, receivedAt time.Time) {
+func (p *Peer) SetHead(hash common.Hash, number *big.Int, entropy *big.Int, receivedAt time.Time) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
 	copy(p.head[:], hash[:])
-	p.number = number
 	p.receivedHeadAt = receivedAt
+	p.number = new(big.Int).Set(number)
+	p.entropy = entropy
```

Comment:
Shouldn't entropy also be copied into a new(big.Int) since it's a pointer? Or what was the purpose of doing that for p.number.

Reply:
yes it should be

Reply:
changed it